### PR TITLE
Exclude preprocessing & imputation steps from scikit-learn autologging

### PR DESCRIPTION
**Pull request by dbczumar**
## What changes are proposed in this pull request?

This PR adjusts the scope of scikit-learn autologging estimators to exclude preprocessing and imputation routines used for data manipulation, label encoding, etc. These routines are not ML models and are often leveraged independently of ML training routines. Accordingly, we should not be capturing MLflow runs for preprocessing & imputation steps. 

We note that preprocessing routines will still be reflected as part of ML pipelines trained via `sklearn.pipeline.Pipeline`.

For reference, the following preprocessing / imputation estimators will no longer capture MLflow runs during autologging:

```
 sklearn.preprocessing._data.Binarizer,
 sklearn.preprocessing._function_transformer.FunctionTransformer,
 sklearn.preprocessing._discretization.KBinsDiscretizer,
 sklearn.preprocessing._data.KernelCenterer,
 sklearn.preprocessing._label.LabelBinarizer,
 sklearn.preprocessing._label.LabelEncoder,
 sklearn.preprocessing._data.MaxAbsScaler,
 sklearn.preprocessing._data.MinMaxScaler,
 sklearn.preprocessing._label.MultiLabelBinarizer,
 sklearn.preprocessing._data.Normalizer,
 sklearn.preprocessing._encoders.OneHotEncoder,
 sklearn.preprocessing._encoders.OrdinalEncoder,
 sklearn.preprocessing._data.PolynomialFeatures,
 sklearn.preprocessing._data.PowerTransformer,
 sklearn.preprocessing._data.QuantileTransformer,
 sklearn.preprocessing._data.RobustScaler,
 sklearn.preprocessing._data.StandardScaler
 sklearn.impute._iterative.IterativeImputer,
 sklearn.impute._knn.KNNImputer,
 sklearn.impute._base.MissingIndicator,
 sklearn.impute._base.SimpleImputer
```

## How is this patch tested?

Included unit tests

## Release Notes

Scikit-learn autologging no longer records MLflow runs for preprocessing steps (e.g., normalization and label encoding)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Should we also exlude the imputer classes: https://scikit-learn.org/stable/modules/classes.html#module-sklearn.impute?

File: tests/sklearn/test_sklearn_autolog.py [hunk 1]
```diff
@@ -1002,3 +1002,32 @@ def test_autolog_configuration_options(log_input_example, log_model_signature):
...
+    (e.g., normalization, label encoding) rather than ML models, do not produce
+    runs when their fit_* operations are invoked independently of an ML pipeline
+    """
+    mlflow.sklearn.autolog()
+
+    # Create a run using the MLflow client, which will be resumed via the fluent API,
+    # in order to avoid setting fluent-level tags (e.g., source and user). Suppressing these
+    # tags simplifies test validation logic
+    client = mlflow.tracking.MlflowClient()
+    run_id = client.create_run(experiment_id=0).info.run_id
```
Comment (harupy) [inline tests/sklearn/test_sklearn_autolog.py:1020]: nice trick :)

**Comment by dbczumar** [comment]
> @dbczumar
> Should we also exlude the imputer classes: https://scikit-learn.org/stable/modules/classes.html#module-sklearn.impute?

Absolutely! This is a great catch. I've updated the PR to exclude imputer classes. Let me know if you can take another look!

File: tests/sklearn/test_sklearn_autolog.py (hunk 1 above)
Comment (dbczumar) [reply tests/sklearn/test_sklearn_autolog.py:1020]: thanks!

**Review by harupy — APPROVED** [review]
LGTM!
